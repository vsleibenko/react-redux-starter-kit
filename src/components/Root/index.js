import React from 'react';
import { useDispatch } from 'react-redux';
import { userActions } from '../../redux/actions';

const Root = () => {
  const dispatch = useDispatch();

  const login = () => dispatch(userActions.login.call());

  return (
    <div>
      <div>Root Component</div>
      <button onClick={login}>LOGIN</button>
    </div>
  );
};

export default Root;
