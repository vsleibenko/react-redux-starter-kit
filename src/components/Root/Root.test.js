import React from 'react';
import { render } from 'enzyme';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';

import Root from '.';

let mockStoreInstance = null;
let mockStore = null;
const initialState = {};

beforeAll(() => {
  mockStoreInstance = configureStore();
});

beforeEach(() => {
  mockStore = mockStoreInstance(initialState);
});

describe('Root', () => {
  it('should renders correctly', () => {
    const component = render(<Provider store={mockStore}><Root /></Provider>);

    expect(component).toMatchSnapshot();
  });
});
