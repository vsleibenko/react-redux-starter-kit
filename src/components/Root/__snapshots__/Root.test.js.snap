// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Root should renders correctly 1`] = `
<div>
  <div>
    Root Component
  </div>
  <button>
    LOGIN
  </button>
</div>
`;
